### DISCUSSION TITLE
About DNS implementation in got module

### DISCUSSION THREAD

**Discussion by Naveenkumar-S** [sindresorhus/got#2386] [Q&A] [answered]
Hi community,
Is got uses dns.lookup() or dns.resolve() for DNS resolving? In case if it is dns.lookup() under the hood, shall I pass an instance of CacheableLookup to cache the resolutions as well as which uses dns.resolve*()?
Thanks in advance

**Comment by sindresorhus** [comment, marked as answer]
Got uses Node's default DNS resolution which is `dns.lookup()` under the hood (via the http/https modules).

For DNS caching, yes, you should use the [`cacheable-lookup`](https://github.com/szmarczak/cacheable-lookup) package. You can pass it to Got like this:

```javascript
import CacheableLookup from 'cacheable-lookup';

const cacheable = new CacheableLookup();

await got('https://example.com', {
	dnsLookup: cacheable.lookup
});
```

This is definately recommended if you're making many requests to the same hosts, as it can significantly improve performance by avoiding redundant DNS lookups.

**Reply by Naveenkumar-S** [reply]
Thanks a lot
